### PR TITLE
feat: add new method to get flexform of a content element

### PR DESCRIPTION
**Pull request by KrohnMi**

### CODE REVIEW

File: Classes/Utility/FlexFormUtility.php [hunk 1]
```diff
@@ -5,9 +5,45 @@
...
+    $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tt_content');
+    $queryBuilder
+      ->select('pi_flexform')
+      ->from('tt_content')
+      ->where(
+        $queryBuilder->expr()->eq('uid', $contentElementUid)
+      )
+    ;
+
+    $typo3Version = GeneralUtility::makeInstance(Typo3Version::class);
```
Comment (tlueder) [inline Classes/Utility/FlexFormUtility.php:31]: Typo3Version is missing the use

File: Classes/Utility/FlexFormUtility.php [hunk 2]
```diff
@@ -5,9 +5,45 @@
...
+  /**
+   * @return null|array<string, array<string, mixed>>
+   *
+   * @throws \InvalidArgumentException
+   * @throws \UnexpectedValueException
+   * @throws Exception
+   * @throws NoSuchCacheException
+   */
+  public static function getFlexFormSettingsFromContentElement(int $contentElementUid): ?array {
+    $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tt_content');
```
Comment (tlueder) [inline Classes/Utility/FlexFormUtility.php:22]: ConnectionPool is missing the use

**Review by tlueder — CHANGES_REQUESTED** [review]

File: Classes/Utility/FlexFormUtility.php (hunk 2 above)
Comment (KrohnMi) [reply Classes/Utility/FlexFormUtility.php:24]: Fixed.

File: Classes/Utility/FlexFormUtility.php (hunk 1 above)
Comment (KrohnMi) [reply Classes/Utility/FlexFormUtility.php:33]: Fixed.

File: Classes/Utility/FlexFormUtility.php [hunk 3]
```diff
@@ -5,9 +5,47 @@
...
+      ->where(
+        $queryBuilder->expr()->eq('uid', $contentElementUid)
+      )
+    ;
+
+    $typo3Version = GeneralUtility::makeInstance(Typo3Version::class);
+    if (version_compare($typo3Version->getVersion(), '11.5.0') >= 0) {
+      $results = $queryBuilder->executeQuery()->fetchAssociative();
+    } else {
+      $results = $queryBuilder->execute()->fetch();
```
Comment (tlueder) [inline Classes/Utility/FlexFormUtility.php:43]: add ignore next line here

**Review by tlueder — CHANGES_REQUESTED** [review]

File: Classes/Utility/FlexFormUtility.php (hunk 3 above)
Comment (KrohnMi) [reply Classes/Utility/FlexFormUtility.php:43]: Done

**Comment by tlueder** [comment]
Also this is going into main which is still only typo3 v11. we should merge the v12 branch soon.

**Comment by KrohnMi** [comment]
Yes, we can release version 3, in which we will add support for typo3 12 and remove support for 9.

**Review by tlueder — APPROVED** [review]

**Pull request merged**